Code/Framework/AzFramework/AzFramework/Archive/Archive.cpp

```diff
@@ -363,6 +364,23 @@ namespace AZ::IO
         , m_mainThreadId{ AZStd::this_thread::get_id() }
     {
         CompressionBus::Handler::BusConnect();
+
+        // If the settings registry is not available at this point,
+        // then something catastrophic has happened in the application startup.
+        // That should have been caught and messaged out earlier in startup.
+        if (auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr)
+        {
+            // Automatically register the event if it's not registered, because
+            // this system is initialized before the settings registry has loaded the event list.
+            AZ::ComponentApplicationLifecycle::RegisterHandler(
+                *settingsRegistry, m_componentApplicationLifecycleHandler,
+                [this](AZStd::string_view /*path*/, AZ::SettingsRegistryInterface::Type /*type*/)
+                {
+                    OnSystemEntityActivated();
+                },
+                "SystemEntityActivated",
+                /*autoRegisterEvent*/ true);
```

Comment:
It seems that `SystemComponentsActivated` would be a better event here as it documents that action that is being waited on is the SystemComponents on the SystemEntity and ModuleEntities have all activated and therefore have reflected their classes to the SerializeContext.

Reply:
I've updated it to use that event and removed the new one.